src/null-safety/faq.md

```diff
@@ -0,0 +1,292 @@
+---
+title: Null safety frequently asked questions
+description: FAQs for Dart's null safety migration
+toc: true
+---
+
```

Comment:
We should add a brief intro here, with a link to /null-safety (at least).

We might also want to mention that this is based on an internal doc that's based on the experience of migrating Google internal code. (I'm not sure about this, but it makes me feel better about not enforcing our editorial style. :)